### ISSUE TITLE
hide cmd window

### ISSUE THREAD

**Issue by NewUserHa** [altendky/pyqt5-tools#32]
hide the cmd windows for convenient.
thanks

**Comment by altendky** [comment]
I think this is simple, just `gui_scripts` rather than `console_scripts`.  But Windows is terrible.  If you do that then even when you run from an existing terminal you won't get to see any output.  This means you lose `--help` and any diagnostic output.  So I understand the desire and share the interest in a tidy experience but given how finicky `pyqt5-tools` is in practice etc I'm hesitant to change this right now.  I'll leave this open though in case I (or someone else) can come up with a good plan to help with this.  A simple one would be to duplicate the entry_points under both console and GUI types with a little different name or such.

**Comment by NewUserHa** [comment]
I found:
1. pyqt5designer.exe or designer.exe is ok without console windows after opened
2. orange-canvas.exe can archieve both `run without console` and `return help if -h`. hope you find out how they implement.

**Comment by altendky** [comment]
So what programs are you getting console windows for if not `pyqt5designer.exe` and `designer.exe`?  And do you have a link for `orange-canvas.exe`?

**Comment by NewUserHa** [comment]
not sure if orange has that ability now because it's in anaconda.
`orange3` from anaconda.

pyshell.exe has a window which is a blocking point of its gui program. but `pyqt5designer.exe and designer.exe` is created by subprocess.

**Comment by altendky** [comment]
https://github.com/biolab/orange3/blob/64e340fe431f4c65c20231b50d7619dce37ef67c/setup.py#L93-L95

I'll try to find time to try this out and see what it does.  But I'm still not clear which programs are doing the thing you dislike.

**Comment by NewUserHa** [comment]
hm? what programs doing dislike?

**Comment by altendky** [comment]
You have reported an issue here.  Something about some console windows you would rather not have.  The only particular programs you mentioned (from `pyqt5-tools`) are `pyqt5designer.exe` and `designer.exe` about which you said: "ok without console windows after opened".  I read that as those two `.exe` files do not have a problem.  So, I don't understand which programs you are having a problem with.

**Comment by NewUserHa** [comment]
oh, I thought "it's ok without console window~~s~~ after it opened" **could mean** 'it's safe to close console window after it opened'

`pyqt5designer.exe` and `designer.exe` will leave a useless console window after `qt designer` opened by double-clicking.
and it's safe to close the window. so I say it's better not to show that window at all.

**Comment by altendky** [comment]
I agree that when launching by double clicking a file it would be nice to not have the console window.  It would also be nice to have console output in the shell, for `--help` and for diagnostics, when run from a terminal.  I'll look at orange3 to observe what it does and if it can handle both I'll see if I can figure out how.

**Comment by NewUserHa** [comment]
not very sure orange3 can do it!

also I saw some program use .py and .pyw to distinguish this problem
